src/RunScriptCommand.cs

```diff
@@ -43,39 +42,160 @@ public async Task<int> InvokeAsync(InvocationContext context)
+        var scripts = context.ParseResult.GetValueForArgument(GlobalArguments.Scripts);
+
+        IConsoleWriter writer = new ConsoleWriter(context.Console, _consoleFormatProvider, verbose);
 
         writer.VerboseBanner();
 
-        var scriptShell = context.ParseResult.GetValueForOption(GlobalOptions.ScriptShell);
-        var (shell, isCmd) = GetScriptShell(scriptShell ?? _project.ScriptShell);
+        Project? project;
+        try
+        {
+            Environment.SetEnvironmentVariable("INIT_CWD", _workingDirectory);
 
-        var ct = context.GetCancellationToken();
+            (project, _workingDirectory) = await new ProjectLoader().LoadAsync(_workingDirectory);
+        }
+        catch (Exception ex)
+        {
+            writer.Error(ex.Message);
 
-        foreach (var script in _scriptNames.Where(scriptName => _project.Scripts!.ContainsKey(scriptName)))
+            return 1;
+        }
+
+        (scriptShell, var isCmd) = GetScriptShell(scriptShell ?? project.ScriptShell);
+
+        writer.LineVerbose("Using shell: {0}", scriptShell);
+        writer.BlankLine();
+
+        if (scripts.Length == 0)
         {
-            var args = script == Name
-                ? context.ParseResult.UnparsedTokens
-                : null;
+            PrintAvailableScripts(writer, project.Scripts!);
 
-            var result = await RunScriptAsync(
+            return 0;
+        }
+
+        // The `env` script is special so if it's not explicitly declared we act like it was
+        var scriptsToRun = scripts
+            .Select(script => (name: script, exists: project.Scripts!.ContainsKey(script) || script == "env"))
+            .ToList();
+
+        // When `--if-present` isn't specified and a script wasn't found in the config then we show an error and stop
+        if (scriptsToRun.Any(s => !s.exists) && !ifPresent)
+        {
+            writer.Error(
+                "Script not found: {0}",
+                string.Join(
+                    ", ",
+                    scriptsToRun
+                        .Where(script => !script.exists)
+                        .Select(script => script.name)));
+
+            return 1;
+        }
+
+        var runResults = new List<(string scriptName, int exitCode)>();
+
+        foreach (var (scriptName, scriptExists) in scriptsToRun)
+        {
+            if (!scriptExists)
+            {
+                writer.Banner($"Skipping script {scriptName}");
+
+                continue;
+            }
+
+            var result = await ProcessScriptAsync(
+                project.Scripts!,
                 writer,
-                script,
-                _project.Scripts![script],
-                shell,
+                scriptShell!,
                 isCmd,
-                args,
-                ct);
+                scriptName,
+                context.ParseResult.UnparsedTokens,
+                context.GetCancellationToken());
+
+            runResults.Add((scriptName, result));
```

Comment:
Tracking the run results like this is overkill when running in series, but we'll need it when doing parallel execution so might as well keep it like this.